src/Draco.Compiler/Internal/Syntax/Lexer.cs

```diff
@@ -784,16 +784,31 @@ private bool TryParseTrivia([MaybeNullWhen(false)] out Trivia result)
             result = Trivia.From(TriviaType.Whitespace, this.AdvanceWithText(offset));
             return true;
         }
-        // Line-comment
+        // Comment
         if (ch == '/' && this.Peek(1) == '/')
         {
-            var offset = 2;
-            // NOTE: We use a little trick here, we specify a newline character as the default for Peek,
-            // which means that this will terminate, even if the comment was on the last line of the file
-            // without a line break
-            while (!IsNewline(this.Peek(offset, @default: '\n'))) ++offset;
-            result = Trivia.From(TriviaType.LineComment, this.AdvanceWithText(offset));
-            return true;
+            // Documentation Comment
+            if (this.Peek(2) == '/')
+            {
+                var offset = 3;
+                // NOTE: We use a little trick here, we specify a newline character as the default for Peek,
+                // which means that this will terminate, even if the comment was on the last line of the file
+                // without a line break
+                while (!IsNewline(this.Peek(offset, @default: '\n'))) ++offset;
+                result = Trivia.From(TriviaType.DocumentationComment, this.AdvanceWithText(offset));
+                return true;
+            }
+            // Line Comment
+            else
+            {
+                var offset = 2;
+                // NOTE: We use a little trick here, we specify a newline character as the default for Peek,
+                // which means that this will terminate, even if the comment was on the last line of the file
+                // without a line break
+                while (!IsNewline(this.Peek(offset, @default: '\n'))) ++offset;
+                result = Trivia.From(TriviaType.LineComment, this.AdvanceWithText(offset));
+                return true;
+            }
```

Comment:
I'd suggest factoring out the core logic, you duplicated a "tricky" codeblock, which is not nice. You could do something like so:
```cs
var offset = 2;
var commentType = TokenType.LineComment;
if (this.Peek(2) == '/')
{
    offset = 3;
    commentType = TokenType.DocumentationComment;
}
// Rest of the code...
```